diff -u xloadimage-4.1/debian/changelog xloadimage-4.1/debian/changelog
--- xloadimage-4.1/debian/changelog
+++ xloadimage-4.1/debian/changelog
@@ -1,3 +1,10 @@
+xloadimage (4.1-16.3) unstable; urgency=low
+
+  * Non-maintainer upload.
+  * libpng transition.
+
+ -- Nobuhiro Iwamatsu <iwamatsu@nigauri.org>  Wed, 08 Jun 2011 21:50:37 +0900
+
 xloadimage (4.1-16.2) unstable; urgency=low
 
   * Non-maintainer upload.
diff -u xloadimage-4.1/debian/patches/00list xloadimage-4.1/debian/patches/00list
--- xloadimage-4.1/debian/patches/00list
+++ xloadimage-4.1/debian/patches/00list
@@ -19,0 +20,2 @@
+20_patch
+21_patch
only in patch2:
unchanged:
--- xloadimage-4.1.orig/debian/patches/20_patch.dpatch
+++ xloadimage-4.1/debian/patches/20_patch.dpatch
@@ -0,0 +1,60 @@
+#! /bin/sh -e
+## 10_config.c-HOME-fix.dpatch by Austin Donnelly <and1000@debian.org>
+##
+## All lines beginning with `## DP:' are a description of the patch.
+## DP: Gracefully handle lack of HOME env. variable.
+
+if [ $# -ne 1 ]; then
+    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
+    exit 1
+fi
+case "$1" in
+       -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
+       -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
+	*)
+		echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
+		exit 1;;
+esac
+
+exit 0
+
+--- a/png.c	2011-06-08 10:28:55.000000000 +0900
++++ b/png.c	2011-06-08 10:32:20.000000000 +0900
+@@ -75,7 +75,11 @@
+ {
+   debug(" #error ");
+   output_warn( png_ptr, str);
++#if PNG_LIBPNG_VER_MAJOR >= 1 && PNG_LIBPNG_VER_MINOR >= 4
++  longjmp(png_jmpbuf((png_ptr)),1);
++#else
+   longjmp(png_ptr->jmpbuf, 1);	/* return control to outer routine */
++#endif
+ }
+ 
+ 
+@@ -164,7 +168,11 @@
+     png_destroy_read_struct(png_pp, info_pp, end_pp);
+     return 0;
+   }
++#if PNG_LIBPNG_VER_MAJOR >= 1 && PNG_LIBPNG_VER_MINOR >= 4
++  if (setjmp(png_jmpbuf(*png_pp))) {
++#else
+   if (setjmp((*png_pp)->jmpbuf)) {
++#endif
+     /* On error */
+     png_destroy_read_struct(png_pp, info_pp, end_pp);
+     return 0;
+@@ -220,7 +228,12 @@
+     zclose(zinput_file);
+     return 0;
+   }
++
++#if PNG_LIBPNG_VER_MAJOR >= 1 && PNG_LIBPNG_VER_MINOR >= 4
++  if (setjmp(png_jmpbuf(png_ptr))) {
++#else
+   if (setjmp(png_ptr->jmpbuf)) {
++#endif
+     /* On error */
+     freeImage(image);
+     png_destroy_read_struct(&png_ptr, &info_ptr, &end_info);
+
only in patch2:
unchanged:
--- xloadimage-4.1.orig/debian/patches/21_patch.dpatch
+++ xloadimage-4.1/debian/patches/21_patch.dpatch
@@ -0,0 +1,34 @@
+#! /bin/sh -e
+## 10_config.c-HOME-fix.dpatch by Austin Donnelly <and1000@debian.org>
+##
+## All lines beginning with `## DP:' are a description of the patch.
+## DP: Gracefully handle lack of HOME env. variable.
+
+if [ $# -ne 1 ]; then
+    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
+    exit 1
+fi
+case "$1" in
+       -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
+       -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
+	*)
+		echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
+		exit 1;;
+esac
+
+exit 0
+
+--- a/png.c	2011-06-08 21:56:57.000000000 +0900
++++ b/png.c	2011-06-08 21:58:16.000000000 +0900
+@@ -321,7 +321,11 @@
+   case PNG_COLOR_TYPE_GRAY_ALPHA:
+   case PNG_COLOR_TYPE_GRAY:
+     if (bit_depth < 8)
++#if PNG_LIBPNG_VER_MAJOR >= 1 && PNG_LIBPNG_VER_MINOR >= 4
++      png_set_expand_gray_1_2_4_to_8(png_ptr);  /* 1 pixlel 1 byte */
++#else
+       png_set_gray_1_2_4_to_8(png_ptr);  /* 1 pixlel 1 byte */
++#endif
+     image = newRGBImage(width, height, 8);
+     image->rgb.used = 256;
+     for (i = 0; i < 256; i++) {
